File: app/src/main/java/com/zia/gankcqupt_mvp/View/Activity/Page/StartActivity.java
package com.zia.gankcqupt_mvp.View.Activity.Page;

import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;

import com.avos.avoscloud.AVOSCloud;
import com.zia.gankcqupt_mvp.Util.Code;
import com.zia.gankcqupt_mvp.Util.PermissionsUtil;
import com.zia.gankcqupt_mvp.View.Activity.BaseActivity;

public class StartActivity extends BaseActivity {

    private final static String TAG = "StartActivityTest";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AVOSCloud.initialize(StartActivity.this, "xKvysVojCOylmIQtqEVGaYQ3-gzGzoHsz", "qRgNqRCPJt9rRsqkftCzvOS1");
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
        finish();
    }

    @Override
    protected void findWidgets() {

    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        switch (requestCode) {
            case Code.DISK:
                if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) { //同意权限申请
                    toast("申请权限成功！");
                } else { //拒绝权限申请
                    toast("没有获取到权限哦..");
                }
                break;
            default:
                break;
        }
    }

}

File: app/src/main/java/com/zia/gankcqupt_mvp/ui/start/StartActivity.kt
package com.zia.gankcqupt_mvp.ui.start

import android.os.Bundle
import com.zia.gankcqupt_mvp.ui.BaseAppCompatActivity

class StartActivity : BaseAppCompatActivity<IStart>(), StartView {


    private val TAG = StartActivity::class.java.canonicalName

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initDataInBackground()
    }

    override fun initDataInBackground() {
        presenter.initLeanCloud(this@StartActivity)
        presenter.initModel()
        presenter.checkPermission(this@StartActivity)
    }

    override fun createPresenter(): IStart {
        return StartPresenter(this)
    }

    override fun getPermissionCode(): Int {
        return 100
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        presenter.onRequestPermissionsResult(this, requestCode, permissions, grantResults)
    }
}
